playwright/_impl/_assertions.py

```diff
@@ -0,0 +1,539 @@
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._expect_impl(
+            "to.be.empty",
+            FrameExpectOptions(timeout=timeout),
+            None,
+            "Locator expected to be empty",
+        )
+
+    async def not_to_be_empty(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._not.to_be_empty(timeout)
+
+    async def to_be_enabled(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._expect_impl(
+            "to.be.enabled",
+            FrameExpectOptions(timeout=timeout),
+            None,
+            "Locator expected to be enabled",
+        )
+
+    async def not_to_be_enabled(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._not.to_be_enabled(timeout)
+
+    async def to_be_hidden(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._expect_impl(
+            "to.be.hidden",
+            FrameExpectOptions(timeout=timeout),
+            None,
+            "Locator expected to be hidden",
+        )
+
+    async def not_to_be_hidden(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._not.to_be_hidden(timeout)
+
+    async def to_be_visible(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._expect_impl(
+            "to.be.visible",
+            FrameExpectOptions(timeout=timeout),
+            None,
+            "Locator expected to be visible",
+        )
+
+    async def not_to_be_visible(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._not.to_be_visible(timeout)
+
+    async def to_be_focused(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._expect_impl(
+            "to.be.focused",
+            FrameExpectOptions(timeout=timeout),
+            None,
+            "Locator expected to be focused",
+        )
+
+    async def not_to_be_focused(
+        self,
+        timeout: float = None,
+    ) -> None:
+        await self._not.to_be_focused(timeout)
+
+
+def expected_regex(
+    pattern: Pattern, match_substring: bool, normalize_white_space: bool
+) -> ExpectedTextValue:
+    expected = ExpectedTextValue(
+        regexSource=pattern.pattern,
+        matchSubstring=match_substring,
+        normalizeWhiteSpace=normalize_white_space,
+    )
+    if pattern.flags != 0:
+        expected["regexFlags"] = ""
+        if (pattern.flags & int(re.IGNORECASE)) != 0:
+            expected["regexFlags"] += "i"
+        if (pattern.flags & int(re.DOTALL)) != 0:
+            expected["regexFlags"] += "s"
+        if (pattern.flags & int(re.MULTILINE)) != 0:
+            expected["regexFlags"] += "m"
+        assert (
+            pattern.flags
+            & ~(
+                int(re.MULTILINE)
+                | int(re.IGNORECASE)
+                | int(re.DOTALL)
+                | int(re.UNICODE)
```

Comment:
JS has "u" flag - should we pass it as well?

Reply:
We didn't do it in Java, thats why I decided to not doing it and in Python every regexp has /u by default.